python/gstaichi/lang/_fast_caching/src_hasher.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Iterable, Sequence
+
+from pydantic import BaseModel
+
+from . import args_hasher, config_hasher, function_hasher
+from .fast_caching_types import FunctionSourceInfo, HashedFunctionSourceInfo
+from .hash_utils import hash_string
+from .pyside_cache import PysideCache
+
+
+def create_cache_key(kernel_source_info: FunctionSourceInfo, args: Sequence[Any]) -> str | None:
+    """
+    cache key takes into account:
+    - arg types
+    - cache value arg values
+    - kernel function (but not sub functions)
+    - compilation config (which includes arch, and debug)
+    """
+    args_hash = args_hasher.hash_args(args)
+    if args_hash is None:
+        return None
+    kernel_hash = function_hasher.hash_kernel(kernel_source_info)
+    config_hash = config_hasher.hash_compile_config()
+    cache_key = hash_string(f"{kernel_hash}_{args_hash}_{config_hash}")
+    return cache_key
+
+
+class CacheValue(BaseModel):
```

Comment:
What a pydantic model for something that only contains a single attribute? For stricter static typing enforcement? I'm not sure it is worth it for something that has no user-facing interface, but as you want.

Reply:
Ok!